packages/salesforcedx-sobjects-faux-generator/src/generator/fauxClassGenerator.ts

```diff
@@ -72,216 +59,26 @@ export class FauxClassGenerator {
       : '';
   }
 
-  private emitter: EventEmitter;
-  private cancellationToken: CancellationToken | undefined;
-  private result: SObjectRefreshResult;
-  private typingGenerator: TypingGenerator;
-  private declGenerator: DeclarationGenerator;
-
-  constructor(emitter: EventEmitter, cancellationToken?: CancellationToken) {
-    this.emitter = emitter;
-    this.cancellationToken = cancellationToken;
-    this.result = { data: { cancelled: false } };
-    this.typingGenerator = new TypingGenerator();
-    this.declGenerator = new DeclarationGenerator();
-  }
-
-  public async generate(
-    projectPath: string,
-    category: SObjectCategory,
-    source: SObjectRefreshSource
-  ): Promise<SObjectRefreshResult> {
-    this.result = { data: { category, source, cancelled: false } };
-    const sobjectsFolderPath = path.join(
-      projectPath,
-      SFDX_DIR,
-      TOOLS_DIR,
-      SOBJECTS_DIR
-    );
-    const standardSObjectsFolderPath = path.join(
-      sobjectsFolderPath,
-      STANDARDOBJECTS_DIR
+  public generate(output: SObjectRefreshOutput): void {
+    const outputFolderPath = path.join(
+      output.sfdxPath,
+      ...REL_BASE_FOLDER,
+      this.relativePath
     );
-    const customSObjectsFolderPath = path.join(
-      sobjectsFolderPath,
-      CUSTOMOBJECTS_DIR
-    );
-    const typingsFolderPath = path.join(projectPath, SFDX_DIR, ...TYPING_PATH);
-
-    if (
-      !fs.existsSync(projectPath) ||
-      !fs.existsSync(path.join(projectPath, SFDX_PROJECT_FILE))
-    ) {
-      return this.errorExit(
-        nls.localize('no_generate_if_not_in_project', sobjectsFolderPath)
-      );
-    }
-    this.cleanupSObjectFolders(sobjectsFolderPath, category);
-
-    const connection = await Connection.create({
-      authInfo: await AuthInfo.create({
-        username: await ConfigUtil.getUsername(projectPath)
-      })
-    });
-    const describe = new SObjectDescribe(connection);
-
-    let sobjects: string[] = [];
-    try {
-      sobjects = await describe.describeGlobal(category, source);
-    } catch (e) {
-      const err = JSON.parse(e);
-      return this.errorExit(
-        nls.localize('failure_fetching_sobjects_list_text', err.message),
-        err.stack
-      );
+    if (!this.resetOutputFolder(outputFolderPath)) {
+      throw nls.localize('no_sobject_output_folder_text', outputFolderPath);
     }
 
-    if (
-      this.cancellationToken &&
-      this.cancellationToken.isCancellationRequested
-    ) {
-      return this.cancelExit();
-    }
+    const definitions =
```

Comment:
Today we have the option of calling fauxClassGenerator.generate() with the "ALL" option. Are we removing that?

Reply:
@jag-sfdc Support for that behavior can be accomplished through configuration. SObjectTransformFactory handles that by adding two FauxClassGenerators - one for standard and one for custom.